From cb1103762449c0d8097d58c701a06118e417a50e Mon Sep 17 00:00:00 2001
From: Stefan Kalscheuer <stefan@stklcode.de>
Date: Fri, 15 Aug 2025 14:48:54 +0200
Subject: [PATCH] arm64: dts: marvell: specity phy-mode "2500base-x" for
 Methode uDPU

In OpenWRT we used to have a copy of the DTS with "2500base-x" instead
of "sgmii" for both ethernet nodes. Apply this change to the upstream
DTS and omit the full-copy.

Signed-off-by: Stefan Kalscheuer <stefan@stklcode.de>
---
 arch/arm64/boot/dts/marvell/armada-3720-uDPU.dts | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

--- a/arch/arm64/boot/dts/marvell/armada-3720-uDPU.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-uDPU.dts
@@ -8,6 +8,11 @@
 	model = "Methode uDPU Board";
 	compatible = "methode,udpu", "marvell,armada3720", "marvell,armada3710";
 
+	aliases {
+		ethernet0 = &eth0;
+		ethernet1 = &eth1;
+	};
+
 	sfp_eth0: sfp-eth0 {
 		compatible = "sff,sfp";
 		i2c-bus = <&i2c0>;
@@ -37,6 +42,10 @@
 };
 
 &eth0 {
-	phy-mode = "sgmii";
+	phy-mode = "2500base-x";
 	sfp = <&sfp_eth0>;
 };
+
+&eth1 {
+	phy-mode = "2500base-x";
+};
